onnxruntime/core/providers/opencl/opencl_utils.h

```diff
@@ -0,0 +1,412 @@
+  }
+
+  template <typename T, typename E = std::is_convertible<T, cl_int>>
+  KernelLauncher& setInt3(T v1, T v2, T v3) {
+    cl_int3 tmp{{static_cast<cl_int>(v1), static_cast<cl_int>(v2), static_cast<cl_int>(v3)}};
+    SKIP_IF_ERRORED(clSetKernelArg(kernel_, index_, sizeof(tmp), &tmp));
+    index_ += 1;
+    return *this;
+  }
+
+  template <typename T, typename E = std::is_convertible<T, cl_int>>
+  KernelLauncher& setInt4(T v1, T v2, T v3, T v4) {
+    cl_int tmp[4] = {static_cast<cl_int>(v1), static_cast<cl_int>(v2), static_cast<cl_int>(v3), static_cast<cl_int>(v4)};
+    SKIP_IF_ERRORED(clSetKernelArg(kernel_, index_, sizeof(tmp), tmp));
+    index_ += 1;
+    return *this;
+  }
+
+  template <typename T>
+  KernelLauncher& setArg(const T& arg) {
+    SKIP_IF_ERRORED(clSetKernelArg(kernel_, index_, sizeof(T), &arg));
+    index_ += 1;
+    return *this;
+  }
+
+  KernelLauncher& setBuffer(cl_mem arg) {
+    SKIP_IF_ERRORED(clSetKernelArg(kernel_, index_, sizeof(cl_mem), &arg));
+    index_ += 1;
+    return *this;
+  }
+
+  KernelLauncher& setBuffer(const Tensor& arg) {
+    return setBuffer(CL_BUFFER_FROM_TENSOR(arg));
+  }
+
+  template <typename T1, typename T2>
+  KernelLauncher& setBuffers(T1&& arg, T2&& other) {
+    setBuffer(std::forward<T1>(arg));
+    return setBuffer(std::forward<T2>(other));
+  }
+
+  template <typename T, typename... Ts>
+  KernelLauncher& setBuffers(T&& arg, Ts&&... args) {
+    setBuffer(std::forward<T>(arg));
+    return setBuffers(std::forward<Ts>(args)...);
+  }
+
+  KernelLauncher& setImage2D(cl_mem arg) {
+    SKIP_IF_ERRORED(clSetKernelArg(kernel_, index_, sizeof(cl_mem), &arg));
+    index_ += 1;
+    return *this;
+  }
+
+  KernelLauncher& setImage2D(const Tensor& arg) {
+    return setImage2D(CL_IMAGE2D_FROM_TENSOR(arg));
+  }
+
+  template <typename T1, typename T2>
+  KernelLauncher& setImage2Ds(T1&& arg, T2&& other) {
+    setImage2D(std::forward<T1>(arg));
+    return setImage2D(std::forward<T2>(other));
+  }
+
+  template <typename T, typename... Ts>
+  KernelLauncher& setImage2Ds(T&& arg, Ts&&... args) {
+    setImage2D(std::forward<T>(arg));
+    return setImage2Ds(std::forward<Ts>(args)...);
+  }
+
+  Status Launch(const OpenCLExecutionProvider& exec, const NDRange& global, const NDRange& local = {});
+
+ private:
+  inline std::string GetKernelFunctionName() {
+    size_t ret_size;
+    clGetKernelInfo(kernel_, CL_KERNEL_FUNCTION_NAME, 0, nullptr, &ret_size);
+    std::string ret(ret_size, '\0');
+    clGetKernelInfo(kernel_, CL_KERNEL_FUNCTION_NAME, ret.size(), ret.data(), nullptr);
+    return ret;
+  }
+
+#undef SKIP_IF_ERRORED
+  cl_kernel kernel_;
+  cl_uint index_;
+  cl_int err_;
+  cl_uint err_index_;
+};
+
+inline size_t HashCombine(size_t a, size_t b) {
+  // https://github.com/boostorg/functional/blob/c839796c8/include/boost/functional/hash/hash.hpp#L256
+  b ^= b + 0x9e3779b9 + (a << 6) + (a >> 2);
+  return b;
+}
+
+// FIXME: this is debug utiltity add by Jicheng Wen
```

Comment:
what needs to be fixed here?

Reply:
This is some debugging utiltity, it should be removed and is removed now

Reply:
Removed